### DISCUSSION TITLE
Editable settings type component

### DISCUSSION THREAD

**Discussion by pamelahu** [twilio-labs/paste#3259] [Office hours] [answered]
<img width="678" alt="Screenshot 2023-06-01 at 4 35 11 PM" src="https://github.com/twilio-labs/paste/assets/18605252/52b8cfc3-eb46-4c1f-99b9-6fc9f571756f">

@tonge495 (Eileen Tong) and I are working on this One Admin page. Currently display pills are used to signify content that is non-editable, editable, linked, linked + editable. We realize this can be confusing for users to decipher (which are editable vs not editable) but we like the compact nature of the component.

We're wondering if you have a solution for how to display these 4 different types of content in a more compact manner than just input field components? Eileen tried with input fields and it got very busy.

For the editable fields, we want them to be selected from a dropdown/date picker.

Ultimately this use case is helpful when there's a lot of settings that need to be edited or displayed or linked on the same page. This will happen a decent bit in One Admin. 

Non-editable display pills
- Created on [date]
- Product
- Brand
- Location
- Requester
- Modified on [date]
- Modified by

Editable display pills
- Status
- Pricing update
- Expected to send on [date]

Linked display pills
- Slack channel

Editable & linked display pills
- Link

**Comment by pamelahu** [comment]
<img width="525" alt="Screenshot 2023-06-01 at 4 44 00 PM" src="https://github.com/twilio-labs/paste/assets/18605252/ef5dd130-ab4c-4580-aa99-0948d64f6bf2">

This is an example of the type of component we were hoping to achieve (inspired by Jira) but this also isn't clearly editable....

**Comment by loreina** [comment]
Hey @pamelahu, can you share a Figma link and/or Loom video? That'll help us understand the intended interaction and we can get back to you shortly, thanks!

**Comment by pamelahu** [comment]
@loreina https://www.figma.com/proto/e242Xtp1dPvWNL67T2onZw/Communications-Center?type=design&node-id=223-159796&scaling=min-zoom&page-id=223%3A141649&starting-point-node-id=223%3A159796

**Comment by pamelahu** [comment]
![Screenshot 2023-06-01 at 4 51 37 PM](https://github.com/twilio-labs/paste/assets/18605252/9c5cd6fd-5519-4242-84ba-4ec2b738d321)

For example, we're hoping that when you click the component... a drop down will appear so the component will change. 

This probably isn't how display pills were attended to be used - but wondering if y'all have any other ideas to achieve this effect (that aren't input fields.)

**Reply by loreina** [reply]
We have a MenuBadge component coming in our release at the end of June that would be great to use here. We can follow up when that's available in Figma to use!

**Comment by pamelahu** [comment]
<img width="752" alt="Screenshot 2023-06-01 at 4 54 14 PM" src="https://github.com/twilio-labs/paste/assets/18605252/5fdcd62f-223d-4e41-80aa-fade4dd8b88c">

^For this example, we want the link to be editable and the link to be clickable once there is a link added.

**Comment by loreina** [comment]
Hi @pamelahu,

Do you want to come by [office hours](https://docs.google.com/spreadsheets/d/10vD91zdwr9hjgDOV3HSGnoC0LgvYw-thmn3X_kvW0xk/edit#gid=664218049) this week? We can tackle all these questions in one go.

**Reply by tonge495** [reply]
@loreina I would like to attend as well, if that works for everyone!

**Comment by pamelahu** [comment]
Yes! See you Thursday for office hours!

**Comment by nkrantz** [comment, marked as answer]
Hi @pamelahu and @tonge495!  [Here's the recording from yesterday's Office Hours](https://www.loom.com/share/1cb305e50fce4e11a4787190f224e914) (password in #help-design-system Slack).

Please follow up with any further explorations or questions!

**Reply by tonge495** [reply]
Thank you!  Appreciate the discussion, will definitely let you know if we have anything else.
